### PR TITLE
Cell Destabilization Fix for No Charge

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl: by Surrealistik
tweak: Retooled power cell detonation to scale with charge more appropriately.
fix: Fixed depleted cells not detonating properly.
tweak: Fuse timer on a destabilized power cell may be as low as 0.3 seconds; there is risk of premature detonation.
tweak: Destabilized power cells immediately detonate when power is drawn from them, including when loaded into a lasgun.
tweak: Destabilized power cells now have a lightning overlay that makes their state obvious.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Would this be necessary if there wasn't a minimum value here associated to the explosion?

**Comment by DominikPanic** [comment]
This doesn't change throw mode at all

**Comment by Rohesie** [comment]
Can't this still produce a (0, 0, 0) infinite explosion?

**Comment by Surrealaser** [comment]
@Rohesie Not per the code as it exists, but I'll re-test after work tonight.

**Comment by Rohesie** [comment]
```

**Comment by Rohesie** [comment]
```

explosion(T, devastation_range, heavy_impact_range, light_impact_range, flash_range)
```

File: code/modules/projectiles/updated_projectiles/guns/energy.dm [hunk 1]
```diff
@@ -385,10 +385,8 @@
 			playsound(user, active_attachable.activation_sound, 15, 1)
 			active_attachable.activate_attachment(src, null, TRUE)
 
-	if(!cell || cell.charge - charge_cost < 0)
+	if(!cell || !cell.use(charge_cost))
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/energy.dm]:
```suggestion
	if(!cell?.use(charge_cost))
```

**Comment by DominikPanic** [comment]
@spookydonut QDEL_IN is the right solution here, correct?

**Comment by spookydonut** [comment]
if this is working fine then merge it

**Pull request merged**